.github/workflows/ci-pipeline.yml

```
@@ -17,6 +17,45 @@ jobs:
     steps:
       - name: Checkout
         uses: actions/checkout@v2
-
+  
       - name: Run tests
         run: docker-compose -f docker-compose.test.yml up --build
+  deploy:
+    name: Deploy to Docker Hub and Heroku
+    runs-on: ubuntu-latest
+    needs: test
+    if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/main' }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Login to Docker Hub
+        uses: docker/login-action@v2
+        with:
+          username: cameronmccormack
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+
+      - name: Build production app
+        run: docker build --target prod --tag cameronmccormack/todo-app:${{ github.sha }} .
+
+      - name: Tag image as latest production image
+        run: docker tag cameronmccormack/todo-app:${{ github.sha }} cameronmccormack/todo-app:latest

```

Minor, but this can be done as a single step if you'd prefer as you can apply multiple tags to one build
```suggestion
      - name: Build production app
        run: docker build --target prod --tag cameronmccormack/todo-app:${{ github.sha }} --tag cameronmccormack/todo-app:latest .
```